### ISSUE TITLE
Inner attributes in block expressions get deleted

### ISSUE THREAD

**Issue by matthew-mcallister** [rust-lang-nursery/rustfmt#2073]
If you pass in input like this...
```rust
let x = {
    #![my_attr]
    do_something()
};
let x = #[my_attr] {
    do_something()
};
```
... what you get is this...
```rust
let x = { do_something() };
let x = #[my_attr]
{ do_something() };
```

The second case is just bad formatting due to assuming that the attribute appears on its own line. More concerning is the first case, in which actually meaningful code gets deleted.

**Comment by topecongiro** [comment]
We probably should disallow rewriting a block in a single line if there are any attributes.

**Comment by topecongiro** [comment]
Closed via #2075.
